src/normalizer/chinese.rs

```diff
@@ -1,26 +1,47 @@
 use std::borrow::Cow;
 
-use character_converter::traditional_to_simplified;
+use pinyin::ToPinyin;
 
 use super::{Normalizer, NormalizerOption};
 use crate::detection::{Language, Script};
 use crate::Token;
 
-/// Normalize Chinese characters by converting them into Simplified Chinese characters.
+/// Normalize Chinese characters by converting them into Pinyin characters.
 ///
-/// This Normalizer uses [`character_converter`] internally to normalize the provided token.
+/// This Normalizer uses [`pinyin`] internally to normalize the provided token.
 pub struct ChineseNormalizer;
 
 impl Normalizer for ChineseNormalizer {
     fn normalize<'o>(
         &self,
         mut token: Token<'o>,
-        _options: NormalizerOption,
+        options: NormalizerOption,
     ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
-        if let Cow::Owned(s) = traditional_to_simplified(token.lemma()) {
-            token.lemma = Cow::Owned(s);
+        let mut lemma = String::new();
+
+        // Need to create char_map before converting into Pinyin
+        if options.create_char_map {
+            let mut char_map = Vec::new();
+
+            for c in token.lemma().chars() {
+                let char_len = c.len_utf8() as u8;
+
+                char_map.push((char_len, char_len));
+            }
+
+            token.char_map = Some(char_map);
         }
 
+        for c in token.lemma().chars() {
+            match c.to_pinyin() {
+                Some(converted) => lemma.push_str(converted.plain()),
```

Comment:
```suggestion
                Some(converted) => lemma.push_str(converted.with_tone()),
```
I'd prefer to keep the tone information at the step 😄